### PR TITLE
feat: Add Adaptation Projects' Add Annotations To OData generator prompting

### PR DESCRIPTION
**Pull request by mmilko01**
https://github.com/SAP/open-ux-tools/issues/2119

- Add prompting relevant to Add Annotations to Odata generator of Adaptation Project
- Add CLI command for executing Add Annotations to Odata for Adaptation Projects

### CODE REVIEW

File: packages/inquirer-common/src/types.ts [hunk 1]
```diff
@@ -53,6 +53,10 @@ export interface GuiOptions {
      * Indicate state in the left hand navigation panel in YUI
      */
     breadcrumb?: boolean | string;
+    /**
+     * Indicates the type of the prompt
+     */
+    type?: string;
```
Comment (IainSAP) [inline packages/inquirer-common/src/types.ts:59]: Can you please add this as a restricted template literal type?

**Review by IainSAP — CHANGES_REQUESTED** [review]

File: packages/adp-tooling/src/prompts/add-annotations-to-odata/index.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                hint: t('prompts.fileSelectOptionTooltip')
+            },
+            when: (answers: AddAnnotationsAnswers) => answers.id !== ''
+        } as ListQuestion<AddAnnotationsAnswers>,
+        {
+            type: 'input',
+            name: 'filePath',
+            message: t('prompts.filePathLabel'),
+            guiOptions: {
+                type: 'file-browser',
```
Comment (IainSAP) [inline packages/adp-tooling/src/prompts/add-annotations-to-odata/index.ts:55]: Please use https://github.com/SAP/open-ux-tools/blob/main/packages/inquirer-common/src/types.ts#L66

**Review by IainSAP — CHANGES_REQUESTED** [review]

File: packages/adp-tooling/src/prompts/add-annotations-to-odata/index.ts (hunk 2 above)
Comment (mmilko01) [reply packages/adp-tooling/src/prompts/add-annotations-to-odata/index.ts:55]: Fixed with: abc2d5cec3f6012f060e64690c499662cc2cda70

File: packages/adp-tooling/src/base/helper.ts [hunk 3]
```diff
@@ -17,3 +24,109 @@ export function isNotEmptyString(input: string | undefined): boolean {
...
+}
+
+/**
+ * Check if a file already exists in a directory.
+ *
+ * @param {string} filePath - The path to the file.
+ * @param {string} checkDirectory - The directory to check.
+ * @returns {boolean} true if the file exists in the directory, false otherwise
+ */
+export function checkDuplicateFile(filePath: string, checkDirectory: string): boolean {
```
Comment (tobiasqueck) [inline packages/adp-tooling/src/base/helper.ts:82]: What is the difference between `checkDuplicateFile` and `checkFileExists` ?

File: packages/create/README.md [hunk 4]
```diff
@@ -26,6 +26,11 @@ Calling `sap-ux add html` will add html files for local preview and testing to t
 ```sh
 sap-ux change add html ui5-test.yaml
 ```
+### annotations-to-odata
```
Comment (tobiasqueck) [inline packages/create/README.md:29]: Please rename it to just `annotations`. `sap-ux add annotations` is totally fine. OData is a give in the context of Fiori apps.

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
Review: added two inline comments
Local test: creating a new annotation file results in a completely empty file, could you please instead create already a skeleton like https://github.com/SAP/open-ux-tools/blob/main/packages/odata-service-writer/templates/add/annotation.xml. That at least has references to the main vocabularies that a developer usually uses.

File: packages/inquirer-common/src/types.ts (hunk 1 above)
Comment (mmilko01) [reply packages/inquirer-common/src/types.ts]: Fixed with: e192056901a826ecd83af9c45188ea1c68231240

**Comment by mmilko01** [comment]
> Review: added two inline comments Local test: creating a new annotation file results in a completely empty file, could you please instead create already a skeleton like https://github.com/SAP/open-ux-tools/blob/main/packages/odata-service-writer/templates/add/annotation.xml. That at least has references to the main vocabularies that a developer usually uses. (The part of the template that is dynamic would require you to connect to the backend and to get the namespace. You don't need to do it here, just don't add it)

@hristotsolev could you please comment here? The current implementation creates an empty `xml` file, do you see any problem in providing a skeleton instead as suggested?

**Comment by mmilko01** [comment]
> Review: added two inline comments Local test: creating a new annotation file results in a completely empty file, could you please instead create already a skeleton like https://github.com/SAP/open-ux-tools/blob/main/packages/odata-service-writer/templates/add/annotation.xml. That at least has references to the main vocabularies that a developer usually uses. (The part of the template that is dynamic would require you to connect to the backend and to get the namespace. You don't need to do it here, just don't add it)

@hristotsolev 

> Review: added two inline comments Local test: creating a new annotation file results in a completely empty file, could you please instead create already a skeleton like https://github.com/SAP/open-ux-tools/blob/main/packages/odata-service-writer/templates/add/annotation.xml. That at least has references to the main vocabularies that a developer usually uses. (The part of the template that is dynamic would require you to connect to the backend and to get the namespace. You don't need to do it here, just don't add it)

@tobiasqueck we discussed with Hristo and pointed out that the reason we are creating an empty file is that we expect the user to copy/paste the content inside. As far as we understand there is an Annotations modeler tool, that can be used to generate the content of this file. Overall, we are ok with adding the skeleton you proposed, but we are not sure if it will be helpful to the user. Also, if we decide to go with the skeleton you propose, can I reuse your example (without the dynamic part) as I don't completely understand the content and I'm not sure if it is a `dummy` content.

File: packages/create/README.md (hunk 4 above)
Comment (mmilko01) [reply packages/create/README.md]: Renamed with: 5a50c963a83e1bb04fb9fabce3b9dec051eb587a

File: packages/adp-tooling/src/base/helper.ts (hunk 3 above)
Comment (mmilko01) [reply packages/adp-tooling/src/base/helper.ts]: After one more look It really seems they are the same :) I refactored the code with: 60ba31705b0cfeb8c22274312dc0cdcbe3de5ae6

File: packages/adp-tooling/src/base/helper.ts [hunk 5]
```diff
@@ -1,19 +1,97 @@
...
 /**
- * Checks if the input is a valid SAP client.
+ * Check if the file exists.
+ *
+ * @param {string} filePath - The path to the file.
+ * @returns {boolean} true if the file exists, false otherwise
+ */
+export function checkFileExists(filePath: string): boolean {
+    return existsSync(filePath);
+}
```
[the remark this answers is not in the document]
Comment (mmilko01) [reply packages/adp-tooling/src/base/helper.ts]: Fixed with: 662d64916ee1ccaf365a9e5a41e9a46279350881

**Comment by tobiasqueck** [comment]
>  As far as we understand there is an Annotations modeler tool, that can be used to generate the content of this file. Overall, we are ok with adding the skeleton you proposed, but we are not sure if it will be helpful to the user. Also, if we decide to go with the skeleton you propose, can I reuse your example (without the dynamic part) as I don't completely understand the content and I'm not sure if it is a dummy content.

The annotation model won't be of much help in an empty file. The skeleton I asked you to add, contains the references to the vocabularies required for creating UI annotations, the annotation modeler uses these.
Yes, you should reuse the one I referred to without the dynamic part.
I'd strongly recommend that you also try it out, add an annotation file and then use the Fiori tools annotation modeler extension to e.g. change the table columns shown in a list report.

**Comment by mmilko01** [comment]
> The annotation model won't be of much help in an empty file. The skeleton I asked you to add, contains the references to the vocabularies required for creating UI annotations, the annotation modeler uses these. Yes, you should reuse the one I referred to without the dynamic part. I'd strongly recommend that you also try it out, add an annotation file and then use the Fiori tools annotation modeler extension to e.g. change the table columns shown in a list report.

Added with: 22b9dedeb7891523383ccecfc9637e6a4a04559c

**Review by tobiasqueck — APPROVED** [review]
* changeset ok
* code is well structured and understandable
* tested locally after (https://github.com/SAP/open-ux-tools/pull/2083/commits/22b9dedeb7891523383ccecfc9637e6a4a04559c) but not the final PR and it worked as expected
* test coverage good

**Review by IainSAP — APPROVED** [review]

**Review by testojs — APPROVED** [review]
* Code looks good
* Good test coverage
* Not tested manually - @nnikovSAP will do extensive manual testing

**Pull request merged**